src/calendar/__tests__/Calendar.test.tsx

```diff
@@ -0,0 +1,170 @@
+import * as React from "react";
+import { subWeeks, addWeeks } from "date-fns";
+import { axe, render, press } from "reakit-test-utils";
+
+import {
+  CalendarCell,
+  CalendarGrid,
+  CalendarHeader,
+  CalendarButton,
+  useCalendarState,
+  CalendarWeekTitle,
+  CalendarCellButton,
+  CalendarStateInitialProps,
+  Calendar as CalendarWrapper,
+} from "../index";
+
+export const CalendarComp: React.FC<CalendarStateInitialProps> = props => {
+  const state = useCalendarState(props);
+
+  return (
+    <CalendarWrapper {...state}>
+      <div className="header">
+        <CalendarButton {...state} goto="previousYear">
+          previous year
+        </CalendarButton>
+        <CalendarButton {...state} goto="previousMonth">
+          previous month
+        </CalendarButton>
+        <CalendarHeader {...state} data-testid="current-year" />
+        <CalendarButton {...state} goto="nextMonth" data-testid="next-month">
+          next month
+        </CalendarButton>
+        <CalendarButton {...state} goto="nextYear">
+          next year
+        </CalendarButton>
+      </div>
+
+      <CalendarGrid {...state} as="table" className="dates">
+        <thead>
+          <tr data-testid="weekDays">
+            {state.weekDays.map((day, dayIndex) => {
+              return (
+                <CalendarWeekTitle
+                  {...state}
+                  as="th"
+                  scope="col"
+                  key={dayIndex}
+                  dayIndex={dayIndex}
+                >
+                  <abbr title={day.title}>{day.abbr}</abbr>
+                </CalendarWeekTitle>
+              );
+            })}
+          </tr>
+        </thead>
+        <tbody>
+          {state.daysInMonth.map((week: any[], weekIndex: React.Key) => (
+            <tr key={weekIndex}>
+              {week.map((day: Date, dayIndex: React.Key) => (
+                <CalendarCell {...state} as="td" key={dayIndex} date={day}>
+                  <CalendarCellButton {...state} date={day} />
+                </CalendarCell>
+              ))}
+            </tr>
+          ))}
+        </tbody>
+      </CalendarGrid>
+    </CalendarWrapper>
+  );
+};
```

Comment:
Use the component from stories.

OR create a folder structure like meter component with __examples__ containing tests & stories based on a common component.

Reply:
We don't want to reuse the stories component, I believe that test components should be isolated otherwise if we change something in the stories' component that could possibly break the tests.

Reply:
Okay, if you think so. Valid point !!